fml/platform/fuchsia/message_loop_fuchsia.cc

```diff
@@ -11,28 +11,52 @@
 namespace fml {
 
 MessageLoopFuchsia::MessageLoopFuchsia()
-    : loop_(&kAsyncLoopConfigAttachToCurrentThread) {}
+  : loop_(&kAsyncLoopConfigAttachToCurrentThread),
+    running_(false) {
+  auto status = zx_timer_create(ZX_TIMER_SLACK_LATE, ZX_CLOCK_MONOTONIC, &timer_);
+  FML_CHECK(status == ZX_OK);
+  timer_wait_ = new async::Wait(timer_, ZX_TIMER_SIGNALED);
+
+  auto wait_handler = [&](async_dispatcher_t* dispatcher,
+                        async::Wait* wait,
+                        zx_status_t status,
+                        const zx_packet_signal_t* signal
+                    ) {
+    RunExpiredTasksNow();
+    wait->Begin(loop_.dispatcher());
+  };
+
+  timer_wait_->set_handler(wait_handler);
+  status = timer_wait_->Begin(loop_.dispatcher());
+  FML_CHECK(status == ZX_OK);
+}
 
-MessageLoopFuchsia::~MessageLoopFuchsia() = default;
+MessageLoopFuchsia::~MessageLoopFuchsia() {
+  delete timer_wait_;
```

Comment:
Hint: We strongly prefer RAII wrappers over raw pointers or naked handles. So if you need a non-default destructor just to close handles or chase pointers, there is almost always an opportunity to use an existing RAII utilitiy.